batch/batch/worker/worker.py

```diff
@@ -425,138 +490,271 @@ async def batch_worker_access_token(self):
+                    'inheritable': default_docker_capabilities,
+                    'permitted': default_docker_capabilities,
+                },
+            },
+            'linux': {
+                'namespaces': [
+                    {'type': 'pid'},
+                    {
+                        'type': 'network',
+                        'path': f'/var/run/netns/{self.netns.network_name}',
+                    },
+                    {'type': 'mount'},
+                    {'type': 'ipc'},
+                    {'type': 'uts'},
+                    {'type': 'cgroup'},
+                ],
+                'uidMappings': [],  # Can control host -> container uid/gid mappings here
+                'gidMappings': [],
+                'resources': {
+                    'cpu': {'shares': weight},
+                    'memory': {
+                        'limit': self.spec['memory'],
+                        'reservation': self.spec['memory'],
+                    },
+                    # 'blockIO': {'weight': min(weight, 1000)}, FIXME blkio.weight not supported
+                },
+                'maskedPaths': [
+                    '/proc/asound',
+                    '/proc/acpi',
+                    '/proc/kcore',
+                    '/proc/keys',
+                    '/proc/latency_stats',
+                    '/proc/timer_list',
+                    '/proc/timer_stats',
+                    '/proc/sched_debug',
+                    '/proc/scsi',
+                    '/sys/firmware',
+                ],
+                'readonlyPaths': [
+                    '/proc/bus',
+                    '/proc/fs',
+                    '/proc/irq',
+                    '/proc/sys',
+                    '/proc/sysrq-trigger',
+                ],
+            },
+        }
 
-            if timed_out:
-                self.short_error = 'timed out'
-                raise JobTimeoutError(f'timed out after {self.timeout}s')
+        if self.spec.get('unconfined'):
+            config['maskedPaths'] = []
+            config['readonlyPaths'] = []
+            config['process']['apparmorProfile'] = 'unconfined'
+            config['linux']['seccomp'] = {'defaultAction': "SCMP_ACT_ALLOW"}
 
-            if self.container_status['out_of_memory']:
-                self.short_error = 'out of memory'
+        return config
 
-            if 'error' in self.container_status:
-                self.state = 'error'
-            elif self.container_status['exit_code'] == 0:
-                self.state = 'succeeded'
+    async def _get_in_container_user(self):
+        user = self.image_config['Config']['User']
+        if not user:
+            uid, gid = 0, 0
+        elif ":" in user:
+            uid, gid = user.split(":")
+        else:
+            uid, gid = self._read_user_from_rootfs(user)
+        if uid != 0 or gid != 0:
+            await check_shell(f'chown -R {uid}:{gid} {self.rootfs_path}/home/{user}')
```

Comment:
Why is this necessary? Does the image transfer process not preserve permissions?

Reply:
I got mixed up with which user buildkit was trying to use this directory for. You're right and this isn't actually necessary